daisi/src/cpps/common/cpps_manager.cpp

```diff
@@ -462,13 +452,15 @@ void CppsManager::scheduleMaterialFlow(const SpawnInfo &info) {
   }
 
   if (init_application) {
-    initMF(i);
+    Simulator::ScheduleWithContext(material_flows_.Get(i)->GetId(), Seconds(0),
+                                   &CppsManager::initMF, this, i);
 
-    Simulator::Schedule(MilliSeconds(2), &CppsManager::startMF, this, i);
+    Simulator::ScheduleWithContext(material_flows_.Get(i)->GetId(), MilliSeconds(2),
+                                   &CppsManager::startMF, this, i);
   }
 
-  Simulator::Schedule(MilliSeconds(4000), &CppsManager::executeMaterialFlow, this, i,
-                      info.friendly_name);
+  Simulator::ScheduleWithContext(material_flows_.Get(i)->GetId(), MilliSeconds(4000),
```

Comment:
Maybe there should be a comment with the reasons behind the scheduled constant times since it is not clear why the material flow needs to wait 4 seconds before its execution can start.

Reply:
Good point. I will address this with an upcoming PR.